spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -158,6 +182,150 @@ public Filter[] pushedFilters() {
     return pushedFilters;
   }
 
+  @Override
+  public boolean pushAggregation(Aggregation aggregation) {
+    if (!pushDownAggregate(aggregation)) {
+      return false;
+    }
+
+    AggregateEvaluator aggregateEvaluator;
+    try {
+      List<BoundAggregate<?, ?>> aggregates =
```

Comment:
I'd probably consider adding another method `SparkAggregates` that would covert am entire `Aggregation`. That way, we will be able to simplify this block.

Reply:
Changed. Thanks